S3 ACL File permissions

ShaggyDev opened the pull request:
## Commit Notes
Added functionality that would extend the ability to set file permissions the the file being uploaded to Amazon S3.

Added a couple of properties that will define the necessary values for setting ACL permissions on the s3 upload

## Purpose
We are using this plugin to upload finished streams to s3 and wanted the ability to set the file permissions when it is uploaded since our media should be accessible for public consumption. I realized that this might be helpful for other teams that would like this level of control. This currently only supports setting a single Access Control on the file, but can theoretically be expanded to accept a list of Access Controls to apply.

dongreenmedl committed:
Added functionality that would extend the ability to set file permissions the the file being uploaded to Amazon S3.
Added a couple of properties that will define the necessary values for setting ACL permissions on the s3 upload

rogerlittin reviewed (changes requested):
Hi,

You have the wrong getProperty* methods for the property types (should be getPropertyStr).

dongreenmedl committed:
Modified the getProperties call to use the correct type.
Added missing import for the PutObjectRequest.
Fixed small spelling error in the AccessControlList object creation.

ShaggyDev commented:
Thank you for point that out. I have made the updates as well as made sure to included the necessary fixes for issues I previously found with that original commit but forgot to transfer here. Let me know if you notice anything else. Thanks
